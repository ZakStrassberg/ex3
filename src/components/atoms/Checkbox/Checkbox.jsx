import PropTypes from 'prop-types';
import React from 'react';
import { noop } from 'lodash';

export default function Checkbox({ name, label, checked, onClick }) {
  const input = (
    <input type="checkbox" name={name} value={name} checked={checked} onClick={onClick} />
  );
  return label ? (
    <label htmlFor={name}>
      <span>{label}</span>
      {input}
    </label>
  ) : (
    input
  );
}

Checkbox.propTypes = {
  name: PropTypes.string,
  checked: PropTypes.bool,
  label: PropTypes.string,
  onClick: PropTypes.func,
};

Checkbox.defaultProps = {
  label: null,
  checked: false,
  name: 'checkbox',
  onClick: noop,
};
