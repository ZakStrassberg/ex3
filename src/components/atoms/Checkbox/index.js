import styled from 'styled-components';

export const StyledCheckbox = styled.input.attrs({
  type: 'checkbox',
})``;

export default StyledCheckbox;
