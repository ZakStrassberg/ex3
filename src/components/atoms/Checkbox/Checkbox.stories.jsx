import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import Checkbox from './';

const defaultProps = {
  onClick: action('clicked'),
  name: "checkbox",
};

console.log('hi');

storiesOf('Checkbox', module)
  .add('without label', () => (
    <div>
      <Checkbox {...defaultProps} />
      <Checkbox {...defaultProps} checked />
    </div>
  ))
  .add('with labels', () => (
    <div>
      <div>
        <Checkbox {...defaultProps} label="Unchecked" />
      </div>
      <div>
        <Checkbox {...defaultProps} label="Checked" checked />
      </div>
    </div>
  ));
