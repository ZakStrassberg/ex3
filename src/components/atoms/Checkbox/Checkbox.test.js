import { noop } from 'lodash';
import React from 'react';

import Checkbox from './';

describe('Checkbox', () => {
  it('should shallow render', () => {
    const wrapper = shallow(<Checkbox />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should not have checked=true when unchecked', () => {
    const wrapper = shallow(<Checkbox />);
    expect(wrapper.find('input').props().checked).toBe(false);
    expect(wrapper).toMatchSnapshot();
  });

  it('should change from unchecked to checked when clicked', () => {
    // let checkboxProps = {};
    // const toggleChecked = (value) => {
    //   console.log('!!!!!!');
    //   console.log(value);
    //   checkboxProps.checked = value.currentTarget.checked;
    // };
    // checkboxProps = {
    //   checked: false,
    //   onClick: toggleChecked,
    // };
    // const wrapper = shallow(<Checkbox {...checkboxProps} />);
    // expect(wrapper.find('input').props().checked).toBe(false);
    //
    // wrapper.find('input').simulate('click', {
    //   stopPropagation: noop,
    //   preventDefault: noop,
    //   currentTarget: {
    //     checked: !checkboxProps.checked,
    //   },
    // });
    // expect(wrapper.find('input').props().checked).toBe(true);
  });
});
