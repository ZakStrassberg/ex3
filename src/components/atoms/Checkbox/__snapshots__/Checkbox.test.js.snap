// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should not have checked=true when unchecked 1`] = `
<input
  checked={false}
  name="checkbox"
  onClick={[Function]}
  type="checkbox"
  value="checkbox"
/>
`;

exports[`Checkbox should shallow render 1`] = `
<input
  checked={false}
  name="checkbox"
  onClick={[Function]}
  type="checkbox"
  value="checkbox"
/>
`;
